agents_infra/agents_infra/agents/base.py

```diff
@@ -123,21 +74,14 @@ def __init__(
     @classmethod
     def from_config_file(cls, filename=None, log_path=None):
         """
-        Create an agent from a configuration (.ini) file.
-
-        Parameters:
-            filename (str): Path to configuration file. Default is None.
-            log_path (str): Path to where the log files will be stored.
-                Default is None.
-
-        Returns:
-            ServerAgent: Child instance of ServerAgent.
+        Create an agent from configuration file.
+        Merges global config (ServerAgent.config) with local config.ini.
         """
-        agent = cls()
-
-        agent._parse_config_file(filename)
+        base_config = ServerAgent.config or Config()
```

Comment:
`cls.config` would be slightly better for more flexibility in inheriting classes.